apps/system/test/unit/app_usage_metrics_test.js

```diff
@@ -195,32 +198,53 @@ suite('AppUsageMetrics:', function() {
         })));
       });
 
-      test('invalid apps', function() {
-        assert.ok(!metrics.shouldTrackApp(null));
-        assert.ok(!metrics.shouldTrackApp(invalidApp));
+      test('don\'t track invalid apps', function() {
+        invalidApps.forEach(function(app) {
+          assert.ok(!metrics.shouldTrackApp(app));
+        });
       });
 
       test('doesn\'t record install/uninstall', function() {
-        var recorded = metrics.recordInstall(invalidApp);
-        assert.equal(getInvalidAppUsage().installs, 0);
-        assert.ok(!recorded);
-
-        recorded = metrics.recordInstall(invalidApp);
-        assert.equal(getInvalidAppUsage().uninstalls, 0);
-        assert.ok(!recorded);
+        invalidApps.forEach(function(app) {
+            var recorded = metrics.recordInstall(app);
+            assert.equal(getUsage(metrics, app).installs, 0);
+            assert.ok(!recorded);
+
+            recorded = metrics.recordInstall(invalidApp);
+            assert.equal(getUsage(metrics, app).uninstalls, 0);
+            assert.ok(!recorded);
+        });
       });
 
       test('doesn\'t record invocation/activity', function() {
-        var recorded = metrics.recordInvocation(invalidApp, 10000);
-        assert.equal(getInvalidAppUsage().invocations, 0);
-        assert.equal(getInvalidAppUsage().usageTime, 0);
-        assert.ok(!recorded);
-
-        recorded = metrics.recordActivity(invalidApp, 'activity');
-        assert.equal(Object.keys(getInvalidAppUsage().activities).length, 0);
-        assert.ok(!('activity' in getInvalidAppUsage().activities));
-        assert.ok(!recorded);
+        invalidApps.forEach(function(app) {
+          var recorded = metrics.recordInvocation(app, 10000);
+          var usage = getUsage(metrics, app);
+          assert.equal(usage.invocations, 0);
+          assert.equal(usage.usageTime, 0);
+          assert.ok(!recorded);
+
+          recorded = metrics.recordActivity(app, 'activity');
+          usage = getUsage(metrics, app);
+          assert.equal(Object.keys(usage.activities).length, 0);
+          assert.ok(!('activity' in usage.activities));
+          assert.ok(!recorded);
+        });
       });
+
+      test('valid then invalid still counts valid app', function() {
+        metrics.recordInvocation(app1, 10000);
+        metrics.recordInvocation(privateBrowserApp, 10000);
+
```

Comment:
this test seems like it is just testing that calling recordInvocation () on a private app does not go back and retroactively erase previously recorded data. It would be more interesting if you invoked app1 again here, then looked for two invocations instead of just one.